### PR TITLE
More kw default literals

### PR DESCRIPTION
**Pull request by laborg**
Opening again, this time against master. (My last PR towards a branch that got merged was in turn automatically closed when the branch got merged).

The PR adds a couple of more checks for kw default literal definitions, based upon following assumptions:
 - UInt[8|16|32|64|128]: the number of digits together with the definition style (0x...,0o...,0b...) define the type width
 - Chars are Chars
 - Float32 are float literals contain a "f" 
 - Bools are either 'true' or 'false'

### CODE REVIEW

**Comment by davidanthoff** [comment]
Looks like we have a merge conflict now, could you resolve? @ZacLN does this broadly look good to you?

**Comment by laborg** [comment]
I've rebased the PR.

File: src/linting/checks.jl [hunk 1]
```diff
@@ -712,22 +712,54 @@ end
...
     end
 end
 
+# returns true if a unsigned `literal_value` would create a value of type `uint_type`
+function is_unsigned(uint_type, kind, literal_value)
+    if uint_type in (:UInt8, :UInt16, :UInt32, :UInt64, :UInt128)
+        # don't count 0x, 0o, 0b prefix
+        n = count(x -> x != '_', literal_value) - 2
+        l = (;:UInt8 => 0, :UInt16 => 1, :UInt32 => 2, :UInt64 => 4, :UInt128 => 8)[uint_type]
+        u = (;:UInt8 => 1, :UInt16 => 2, :UInt32 => 4, :UInt64 => 8, :UInt128 => 16)[uint_type]
```
Comment (pfitzseb) [inline src/linting/checks.jl:756]: Nitpick, but move these tuples out of the function?

Comment (laborg) [reply src/linting/checks.jl]: This was over complicated for no reason. In the newest commit I just used `sizeof`.

File: src/linting/checks.jl [hunk 2]
```diff
@@ -707,22 +707,54 @@ end
     check_kw_default(x::EXPR, server)
 
 Check that the default value matches the type for keyword arguments. Currently limited to
-the following types `Union{String, Symbol, Int, Float64}`.
+the following types 
+`Union{String, Symbol, Int, Char, Bool, Float32, Float64, UInt8, UInt16, UInt32, UInt64}`.
 """
 function check_kw_default(x::EXPR, server)
```
[the remark this answers is not in the document]
Comment (laborg) [reply src/linting/checks.jl:718]: Thanks for the suggestions. I've incorporated them and refactored the implementation a little bit. The check now also includes non-native-sized signed integers.

**Pull request merged**